File: CodeWars/6kyu/duplicate-encoder/main.js
// 🦉 My Method:
// prettier-ignore
function duplicateEncode(word) {
    Array.prototype.count = function (element) {
        let count = 0;
        this.forEach((item) => {
            if (item === element) {
                count++;
            }
        });

        return count;
    };
    const wordArray = Array.from(word.toLowerCase());

    return Array.from(word.toLowerCase()).map((element) => {
            return wordArray.count(element) > 1 ? ")" : "(";
        }).join("");
}

const test_case = ["din", "recede", "Success", "(( @"];
test_case.forEach((element, index) => console.log(duplicateEncode(element)));

File: CodeWars/6kyu/duplicate-encoder/main.py
# My Method 1~ 🦉
def duplicate_encode(word):
    l_word = word.lower()
    return "".join([")" if l_word.count(letter) > 1 else "(" for letter in l_word])


test_case = ["din", "recede", "Success", "(( @", "E( cI!ACXAU@fN"]
for test in test_case:
    print(duplicate_encode(test))
